_source/_posts/2022-06-01-cloud-native-java-microservices-with-istio.md

```diff
@@ -0,0 +1,604 @@
+}
+
+enum InvoiceStatus {
+  PAID, ISSUED, CANCELLED
+}
+
+entity Shipment {
+  trackingCode String
+  date Instant required
+  details String
+}
+
+enum PaymentMethod {
+  CREDIT_CARD, CASH_ON_DELIVERY, PAYPAL
+}
+
+relationship OneToMany {
+  Invoice{shipment} to Shipment{invoice(code) required}
+}
+
+service Invoice, Shipment with serviceClass
+paginate Invoice, Shipment with pagination
+microservice Invoice, Shipment with invoice
+
+/**
+ * Entities for notification microservice
+ */
+entity Notification {
+  date Instant required
+  details String
+  sentDate Instant required
+  format NotificationType required
+  userId Long required
+  productId Long required
+}
+
+enum NotificationType {
+  EMAIL, SMS, PARCEL
+}
+
+microservice Notification with notification
+```
+
+We define entities for each service and mark the entities as microservice entities. We also define relationships between entities, enums, and other options like pagination, service layer, etc. Please refer to JDL [Entities](https://www.jhipster.tech/jdl/entities-fields) and [relationships](https://www.jhipster.tech/jdl/relationships) documentation for more possibilities.
+
+Now, we are ready to run JHipster. Open a terminal window on the folder where you saved the JDL and run the below command.
+
+```bash
+jhipster jdl app.jdl --fork
+```
+
+This will create the applications with all their entities and specified configurations. You should be able to see the gateway application in action by running the below command on the **store** folder.
+
+```bash
+./gradlew # starts the spring boot application
+```
+
+You can find a [sample application on GitHub](https://github.com/oktadev/okta-java-spring-k8s-istio-microservices-example).
+
+## Create a GKE Cluster and Install Istio
+
+To deploy the stack to Google Kubernetes Engine, we need to create a cluster and install Istio. So let's begin by creating a cluster using Google Cloud SDK.
+
+### Create a cluster
+
+Ensure you are logged into the gcloud CLI and run the below command to create a GKE cluster.
+
+```bash
+# set region and zone
+gcloud config set compute/region europe-west1
+gcloud config set compute/zone europe-west1-b
+# Create a project and enable container APIs
+gcloud projects create jhipster-demo-okta # You need to also enable billing via GUI
+gcloud config set project jhipster-demo-okta
+gcloud services enable container.googleapis.com
+
+# Create GKE Cluster
+gcloud container clusters create hello-hipster \
+   --num-nodes 4 \
+   --machine-type n1-standard-2
+```
+
+This could take anywhere between 5 to 15 minutes. `--machine-type` is important as we need more CPU than available in the default setup. Once the cluster is created, it should be set automatically as the current Kubernetes context. You can verify that by running `kubectl config current-context`. If the new cluster is not set as the current context, you can set it by running `gcloud container clusters get-credentials hello-hipster`.
```

Comment:
```suggestion
This could take anywhere from 5 to 15 minutes. `--machine-type` is important as we need more CPU than available in the default setup. Once the cluster is created, it should be set automatically as the current Kubernetes context. You can verify that by running `kubectl config current-context`. If the new cluster is not set as the current context, you can set it by running `gcloud container clusters get-credentials hello-hipster`.
```
more idiomatic